src/proof/PGTransformationAlgorithms.cc

```diff
@@ -189,7 +190,7 @@ double ProofGraph::recyclePivotsIter() {
                     assert(n == res->getAnt1() or n == res->getAnt2());
                     if (res->getAnt1() == n) { res->setAnt1(replacing); }
                     else if (res->getAnt2() == n) { res->setAnt2(replacing); }
-                    else opensmt_error_();
+                    else { throw OsmtInternalException("Invalid proof structure " + std::string(__FILE__) + ", " + std::to_string(__LINE__)); }
```

Comment:
This is protected from happening by the assert in line 190.  Should we maybe remove the assert then?

Reply:
I would keep the `assert`, I like asserts, asserts are expressive :)
For me the question is if we should throw the exception, so any problem is caught also in Release and not only in Debug.
Since the proof related code is quite complex, I think it makes sense to keep it as well.

Reply:
Ok!